### PR TITLE
NVMeoF b2b upgrade test suite

### PR DESCRIPTION
**Pull request by sunilkumarn417**
# Description

- NVMeoF Gateway Build-to-build upgrade.

**_`CEPH-83576090 | NVMe Build to Build Direct Upgrade`_**

```
All test logs located here: /Users/sunilkumarn/logs/b2b-upgrade-02

TEST NAME                        TEST DESCRIPTION                                               DURATION                         STATUS                    COMMENTS
setup install pre-requisistes    Setup phase to deploy the required pre-requisites for runnin   0:06:13.781017                   Pass
Bootstrap RHCS 7.x(GA) cluster   Bootstrap RHCS 7.x(GA) cluster and deploy services with labe   0:15:29.433281                   Pass
Configure client1 and client2    Configure the RBD client systems                               0:01:52.713422                   Pass
Upgrade Cluster to latest 7.x    Upgrade cluster to latest version with NVMeoF                  0:21:31.769711                   Pass
NVMeoF 4-GW HA test Single fai   4GW HA test post upgrade                                       0:15:56.282388                   Pass
```

### CODE REVIEW

File: ceph/ceph_admin/__init__.py [hunk 1]
```diff
@@ -138,32 +139,47 @@ def set_tool_repo(self, repo=None):
...
+                "8": "https://public.dhe.ibm.com/ibmdl/export/pub/storage/ceph/ibm-storage-ceph-5-rhel-8.repo",
+            },
+        }
+
+        rh_build = self.config.get("rhbuild", "7.1-rhel-9")
+        _release = rh_build[0]
+        if release:
+            _release = release[0]
+        os_major_version = rh_build.split("-")[-1]
+        ibm_build = self.config.get("ibm_build")
```
Comment (psathyan) [inline ceph/ceph_admin/__init__.py:172]: Thanks for adding `ibm` support and improving the approach of accessing repo information. IMHO, we should have this in a separate commit. 

Instead of having it as code, wdyt about having a separate file holding the CDN data.

File: ceph/ceph_admin/__init__.py [hunk 2]
```diff
@@ -138,32 +139,47 @@ def set_tool_repo(self, repo=None):
             for node in self.cluster.get_nodes():
                 node.exec_command(sudo=True, cmd=cmd)
 
-    def set_cdn_tool_repo(self):
+    def set_cdn_tool_repo(self, release=None):
```
Comment (psathyan) [inline ceph/ceph_admin/__init__.py:142]: Kindly add `doc string` for the new parameter.

File: ceph/ceph_admin/registry_login.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
+"""Cephadm registry login CLI Command."""
```
Comment (psathyan) [inline ceph/ceph_admin/registry_login.py:1]: Likewise, this can be a separate commit as it is introducing a new functionality.

File: ceph/nvmegw_cli/execute.py [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
 class ExecuteCommandMixin:
     BASE_CMD = "podman run --quiet --rm"
-    NVMEOF_CLI_IMAGE = "quay.io/ceph/nvmeof-cli:1.1.0"
+    NVMEOF_CLI_IMAGE = "quay.io/ceph/nvmeof-cli:latest"
```
Comment (psathyan) [inline ceph/nvmegw_cli/execute.py:9]: A separate commit for moving to `latest`. This allows to have meaningful history and also track why it was changed.

File: ceph/ceph_admin/__init__.py (hunk 1 above)
Comment (sunilkumarn417) [reply ceph/ceph_admin/__init__.py:172]: There are another discussin going on for that to bring that in entire framework.

File: ceph/ceph_admin/__init__.py (hunk 2 above)
Comment (sunilkumarn417) [reply ceph/ceph_admin/__init__.py:142]: Done.

File: ceph/ceph_admin/registry_login.py (hunk 3 above)
Comment (sunilkumarn417) [reply ceph/ceph_admin/registry_login.py:1]: Done.

File: ceph/nvmegw_cli/execute.py (hunk 4 above)
Comment (sunilkumarn417) [reply ceph/nvmegw_cli/execute.py:9]: Done.

**Review by vamahaja — COMMENTED** [review]
LGTM

**Review by psathyan — APPROVED** [review]
Let's pick up the movement of CDN / repo information in another PR.
